crates/fuel-core/src/combined_database.rs

```diff
@@ -0,0 +1,92 @@
+use crate::database::{
+    database_description::{
+        off_chain::OffChain,
+        on_chain::OnChain,
+        relayer::Relayer,
+    },
+    Database,
+    Result as DatabaseResult,
+};
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    blockchain::primitives::DaBlockHeight,
+    fuel_types::BlockHeight,
+};
+
+/// A database that combines the on-chain, off-chain and relayer databases into one entity.
+#[derive(Default, Clone)]
+pub struct CombinedDatabase {
```

Comment:
Why do these need to be combined? From the domain perspective, if there is an on-chain and an off-chain db, then they should be handed to the domain separately imo.

Reply:
I do see that it gets separated when you create the graph_ql service, for example. So I figure this is just a convenience thing for moving them around.